huaweicloud/services/rds/resource_huaweicloud_rds_instance.go

```diff
@@ -1086,6 +931,90 @@ func buildRdsInstanceParameters(params *schema.Set) instances.ModifyConfiguratio
+	instanceID string, parametersRaw *schema.Set) error {
+	configOpts := buildRdsInstanceParameters(parametersRaw)
+	retryFunc := func() (interface{}, bool, error) {
+		_, err := instances.ModifyConfiguration(client, instanceID, configOpts).Extract()
+		retry, err := handleMultiOperationsError(err)
+		return nil, retry, err
+	}
+	_, err := common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+		Ctx:          ctx,
+		RetryFunc:    retryFunc,
+		WaitFunc:     rdsInstanceStateRefreshFunc(client, instanceID),
+		WaitTarget:   []string{"ACTIVE"},
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		DelayTimeout: 10 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+	if err != nil {
+		return fmt.Errorf("error modifying parameters for RDS instance (%s): %s", instanceID, err)
+	}
+
+	// Check if we need to restart
+	restart, err := checkRdsInstanceRestart(client, instanceID, parametersRaw.List())
+	if err != nil {
+		return err
+	}
+
+	if restart {
+		return restartRdsInstance(ctx, d, client, instanceID)
+	}
+	return nil
+}
+
+func checkRdsInstanceRestart(client *golangsdk.ServiceClient, instanceID string, parameters []interface{}) (bool, error) {
+	configs, err := instances.GetConfigurations(client, instanceID).Extract()
+	if err != nil {
+		return false, fmt.Errorf("error fetching the instance parameters (%s): %s", instanceID, err)
+	}
+
+	for _, parameter := range parameters {
+		name := parameter.(map[string]interface{})["name"]
+		for _, v := range configs.Parameters {
+			if v.Name == name && v.Restart {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+func restartRdsInstance(ctx context.Context, d *schema.ResourceData, client *golangsdk.ServiceClient,
+	instanceID string) error {
+	// If parameters which requires restart changed, reboot the instance.
+	retryFunc := func() (interface{}, bool, error) {
+		_, err := instances.RebootInstance(client, instanceID).Extract()
+		retry, err := handleMultiOperationsError(err)
+		return nil, retry, err
+	}
+	_, err := common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+		Ctx:          ctx,
+		RetryFunc:    retryFunc,
+		WaitFunc:     rdsInstanceStateRefreshFunc(client, instanceID),
+		WaitTarget:   []string{"ACTIVE"},
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		DelayTimeout: 10 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+	if err != nil {
+		return fmt.Errorf("error rebooting for RDS instance (%s): %s", instanceID, err)
+	}
+
+	// wait for the instance state to be 'ACTIVE'.
+	stateConf := &resource.StateChangeConf{
+		Target:       []string{"ACTIVE"},
+		Refresh:      rdsInstanceStateRefreshFunc(client, instanceID),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        5 * time.Second,
+		PollInterval: 5 * time.Second,
+	}
+	if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+		return fmt.Errorf("error waiting for RDS instance (%s) become active state: %s", instanceID, err)
```

Comment:
```suggestion
		return fmt.Errorf("error waiting for RDS instance (%s) become active status: %s", instanceID, err)
```
in the programming，we always use the `status` word to explain a phase, not `state`.

Reply:
fixed